### PR TITLE
Cart system approval checkout flow

### PR DESCRIPTION
**Pull request by antho1404**
Refactor the cart components and add approval and final batch transaction

Note: Lint can safely be ignored will be fixed at the end when the graphql api is live

### CODE REVIEW

File: components/Sales/Direct/CardFooter.tsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ const SaleDirectCardFooter: FC<Props> = ({
           wordBreak="break-all"
         >
           {showButton
-            ? isOwner
+            ? isOwner // TODO: we can be owner and still purchase for erc1155
```
Comment (antho1404) [inline components/Sales/Direct/CardFooter.tsx:93]: TODO for later, I will transform that into issues

**Review by ismailToyran — APPROVED** [review]
Looking great, put some comments.
Only thing that might be good is that, after purchase completion,
We might add callback to fetch assets on the page as an optional param.
For example I've used cart on explorer fixed price filter on,
After purchase item was still there. So we could refetch these items in this case.
Working great. (Maybe later on as an improvement, we could add balance alert on transaction step to let user they don't have enough balance and disable button as well)

File: components/Cart/Step/Selection.tsx [hunk 2]
```diff
@@ -8,96 +11,228 @@ import {
...
+                      key={cartItemWithOffer.offerId}
+                      offer={cartItemWithOffer.offer}
+                    />
+                  ),
+                )}
+              </CartRadio>
+            )
+          })}
+        </VStack>
+      </DrawerBody>
```
[the remark this answers is not in the document]
Comment (antho1404) [reply components/Cart/Step/Selection.tsx:208]: I don't think this works correctly, your html should only have a unique id, so you cannot reuse selection to both to connect them (or there is something new I'm not aware)
Usually, you do want to wrap everything on your form, so you will have things like pressing enter in a field trigger the button of type submit within this form

File: components/Cart/ItemSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import { useFetchCartItemsQuery } from '../../graphql'
+import { CartItem } from '../../hooks/useCart'
+import useEnvironment from '../../hooks/useEnvironment'
+import Image from '../Image/Image'
+import List, { ListItem } from '../List/List'
+import Price from '../Price/Price'
+
+type Props = {
+  chainId: number
+  cartItems: CartItem[]
```
[the remark this answers is not in the document]
Comment (antho1404) [reply components/Cart/ItemSummary.tsx:14]: It could, I just wanted to make sure that we have the full cart item to later display the quantities

**Comment by antho1404** [comment]
The balance is actually tracked in the PR to merge to https://github.com/liteflow-labs/starter-kit/pull/471, this is definitely something missing

For the refresh, that will be tricky as the explore page doesn't get the callback of a purchase, but I will try to work on something

File: components/Sales/Direct/CardFooter.tsx (hunk 1 above)
Comment (antho1404) [reply components/Sales/Direct/CardFooter.tsx:93]: https://github.com/liteflow-labs/starter-kit/issues/483

**Pull request merged**